src/containers/dashboard/components/Admin/UserListActions.jsx

```diff
@@ -1,74 +1,202 @@
 import React from 'react'
-import { Icon, Tooltip, Modal } from 'antd'
+import { Icon, Tooltip, Modal, Button, Select } from 'antd'
 import { connect } from 'react-redux'
 import { setAdmin, removeAdmin, validatePayment } from '../../../../modules/admin'
 
+import './admin.css'
+
 class UserListActions extends React.Component {
   constructor(props) {
     super(props)
+    
     this.state = {
-      modalVisible: false
+      paymentModalVisible: false,
+      setAdminModalVisible: false,
+      removeAdminModalVisible: false
     }
   }
 
   setAdmin = () => {
     this.props.setAdmin(this.props.userId)
+    this.setState({
+      setAdminModalVisible: false
+    })
   }
 
   removeAdmin = () => {
     this.props.removeAdmin(this.props.userId)
+    this.setState({
+      removeAdminModalVisible: false
+    })
   }
 
-  openModal = () => {
+  validatePayment = () => {
+    this.props.validatePayment(this.props.userId)
     this.setState({
-      modalVisible: true,
+      paymentModalVisible: false
     })
   }
 
-  validate = () => {
-    this.props.validatePayment(this.props.userId)
+  openMainModal = () => {
+    this.setState({
+      mainModalVisible: true
+    })
+  }
+
+  closeMainModal = () => {
+    this.setState({
+      mainModalVisible: false
+    })
+  }
+
+  openSetAdminModal = () => {
     this.setState({
-      modalVisible: false,
+      mainModalVisible: false,
+      setAdminModalVisible: true
     })
   }
 
-  closeModal = () => {
+  closeSetAdminModal = () => {
     this.setState({
-      modalVisible: false,
+      mainModalVisible: true,
+      setAdminModalVisible: false
+    })
+  }
+
+  openRemoveAdminModal = () => {
+    this.setState({
+      mainModalVisible: false,
+      removeAdminModalVisible: true
+    })
+  }
+
+  closeRemoveAdminModal = () => {
+    this.setState({
+      mainModalVisible: true,
+      removeAdminModalVisible: false
+    })
+  }
+
+  openPaymentModal = () => {
+    this.setState({
+      mainModalVisible: false,
+      paymentModalVisible: true
+    })
+  }
+
+  closePaymentModal = () => {
+    this.setState({
+      mainModalVisible: true,
+      paymentModalVisible: false
     })
   }
 
   render() {
     const { users, userId } = this.props
     const user = users.find(u => u.id === userId)
 
+    if (!user) {
+      return null
+    }
+
+    let admin = user.permissions && user.permissions.admin === 100
```

Comment:
Cette page est uniquement accessible par les admins. Je pense que tu peux directement tej l'utilisateur qui n'a rien à faire là vers la page d'accueil

```suggestion
    let admin = user.permissions && user.permissions.admin === 100
    if (!admin){
      this.props.goToHome() // si tu as bien export la fonction via redux
    }
```

Reply:
ok